private/octet-stream.rkt

```diff
@@ -35,38 +36,39 @@
 (define (octet-stream bytes #:padding [padding 0])
   (plain-octet-stream bytes padding))
 
+(define (dissect-octet-stream octets body)
+  (body (octet-stream-bytes octets) (octet-stream-padding octets)))
+
 (define (octet-stream->bitstring octets)
-  (bytes->bitstring (octet-stream-bytes octets)
-                    #:padding (octet-stream-padding octets)))
+  (dissect-octet-stream octets #/fn bytes padding
+  #/bytes->bitstring bytes #:padding padding))
 
 (define (application/octet-stream #:padding [padding 0])
-  (if (zero? padding)
-      (media-type 'application 'octet-stream)
-      (media-type 'application 'octet-stream
-                  #:parameters
-                  (record #:padding (number->immutable-string padding)))))
+  (mat padding 0
+    (media-type 'application 'octet-stream)
+    (media-type 'application 'octet-stream
+                #:parameters
+                (record #:padding #/number->immutable-string padding))))
```

Comment:
If you're not careful, you could accidentally put them in the `(media-type ...)` call, right? Arguments go where they go.

Something like `padding` here isn't an argument to `(record ...)`. It's definitely an argument to `(number->immutable-string ...)`, and I think it also makes sense to say it's an argument to `(record #:padding #/number->immutable-string ...)` or even `(media-type ... #:parameters (record #:padding #/number->immutable-string ...))`, but I wouldn't say it's an argument to `(record ...)` itself or `(media-type ...)` itself.

Anyway, if you're anticipating this is likely to become a record with two or more fields, then I don't think I would use a weak paren here!

I made the choice I did because I judged from the context that the most likely refactoring here would be not to add a new field to the record but instead to define an abstraction for `(record #:padding (number->immutable-string ...))`. There are at least two temptations that can lead to that abstraction: The abstraction can make this code less verbose in terms of raw character count, and it can reduce the paren nesting. By using Parendown to reduce the paren nesting, I'm addressing one of these temptations right now and thus allowing that abstraction (if it ever does exist) to be designed with a purer focus.

This is a preference I picked up using Arc's composition operator `(a:b:c ...)`, which only worked when `a`, `b`, `c` were symbols. By using `(foo:list ...)`, I could pretend I had a varargs variant of `foo` named `foo:list`. By using `(tostring:write ...)`, I could pretend I had a variant of `write` named `tostring:write` that returned its output as a string rather than writing it to an output port. These "names" would often be at least as terse as whatever other names I could come up with for the same behaviors, so I could focus on defining abstractions that served more substantial purposes. My use of `(a /b /c ...)` originated as a generalization of Arc's `(a:b:c ...)`, so even though I've eventually come to think of it as having primarily indentation-related advantages, I still like to use it as a kind of composition operator.